File: python/mouse.py
import pyautogui as mouse
import time
import datetime
datetime.tzinfo = "EST"
now = datetime.datetime.now().time()
start = datetime.time(8)
end = datetime.time(17)
mouse.failsafe = False

while(start.hour <= now.hour <= end.hour):
    starting_position = mouse.position()
    print("sleeping 1 minute")
    time.sleep(60)
    if(mouse.position() != starting_position):
        print("mouse moved to position {}".format(mouse.position()))
        print("sleeping 1 minute")
        time.sleep(60)
    else:
        print("mouse not moving")
        print("moving mouse for you")
        mouse.moveTo(mouse.position().x, mouse.position().y + 100)
        mouse.moveTo(mouse.position().x, mouse.position().y - 100)
        print("pressing shift 3 times")
        mouse.press('shift', presses=3)
        print("movement made at {} to position {}".format(now, mouse.position()))
    time.sleep(10)

File: ruby/mouse.rb
#!/usr/bin/env ruby

require 'bundler/inline'

gemfile do
    source 'https://rubygems.org'
    gem 'rumouse'
end

mouse = RuMouse.new
time = Time.now

until 8 >= time.hour || time.hour <= 17 do
    starting_position = mouse.position
    puts "sleeping 1 minute"
    sleep 60
    if mouse.position != starting_position
        puts "mouse moved to position #{mouse.position}"
        puts "sleeping 30 seconds"
        sleep 30
    else
        puts "mouse not moving"
        puts "moving mouse for you"
        mouse.move mouse.position[:x], (mouse.position[:y]+100)
        sleep 0.1
        mouse.move mouse.position[:x], (mouse.position[:y]-100)
        puts "sleeping 15 minutes from current time: #{time.hour}:#{time.min}"
        sleep 15*60
    end
end


